### PR TITLE
Async debugger documentation

### PR DESCRIPTION
**Pull request by wpopielarski**
Added some docs about Async Debugger

### CODE REVIEW

File: src/sphinx/4.0.x/features/async-debugger/index.rst [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+The Asynchronous Debugger extensions facilitate debugging of applications that use actors, futures or even iteratees. There are two main features:
+
+- Asynchronous Stack Recovery. Capture the call stack at the point where a future is created, or a message is sent to an actor.
+- Step with Message. Follow a message send and stop in the receiving actor.
+
+A few minor improvements, too:
+
+- logical structures for actors (see at a glance the supervisor, current sender, supervising strategy)
+- break on dead letters (breakpoint when a message arrives in dead letters)
```
Comment (sschaef) [inline src/sphinx/4.0.x/features/async-debugger/index.rst]: items in enumerations should start with an upper case letter

File: src/sphinx/4.0.x/features/async-debugger/index.rst [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+.. image:: images/async-stack-5.png
+    :width: 100%
+
+Step with Message
+-----------------
+
+A mysterious green *bang* icon on ``Debug`` view toolbar is a next part of Async Debugger tools suite. When it is clicked and underlying code line launches a message then the code execution is continued up to a place where this message is received. In other case it works like ``step over`` action.
+
+Let's flick thru a simple example. First stop (with breakpoint) in a part of code which emits messages.
```
Comment (sschaef) [inline src/sphinx/4.0.x/features/async-debugger/index.rst]: `s/thru/through/`

**Comment by sschaef** [comment]
Small nitpicks, beside from that LGTM

**Comment by dragos** [comment]
LGTM!

**Pull request merged**